packages/react-native-sdk/src/hooks/useLocalVideoStream.ts

```diff
@@ -18,8 +18,17 @@ export const useLocalVideoStream = () => {
 
   useEffect(() => {
     const loadVideoStream = async () => {
-      const stream = await getVideoStream(currentVideoDevice?.deviceId);
-      setVideoStream(stream);
+      // If there is no video device, we don't need to load a video stream.
+      if (!currentVideoDevice?.deviceId) return null;
+
+      const stream = await getVideoStream({
+        deviceId: currentVideoDevice.deviceId,
+      });
+      setVideoStream((previousStream) => {
+        if (!previousStream) return stream;
+
+        disposeOfMediaStream(previousStream);
+      });
```

Comment:
Currently this effect will never update the stream if the deviceId changes 

```suggestion
      setVideoStream((previousStream) => {
        if (previousStream) disposeOfMediaStream(previousStream);
        return stream;
      });
```